tests/linux/test_backend_utils.cpp

```diff
@@ -238,3 +309,59 @@ TEST(BackendUtils, image_resizing_not_attempted_when_minimum_size_not_understood
     test_image_resizing(img, min_size, request_size, qemuimg_info_output, qemuimg_info_result, attempt_resize,
                         qemuimg_resize_result, throw_msg_matcher);
 }
+
+TEST(BackendUtils, image_convert_returns_same_path_if_not_raw_image)
+{
+    const auto img_path = "/fake/img/path";
+    const auto expected_img_path = img_path;
+    const auto qemuimg_info_output = "{\n    \"format\": \"qcow2\"\n}";
+    const auto qemuimg_info_result = success;
+    const auto attempt_convert = false;
+    const auto qemuimg_convert_result = mp::ProcessState{};
+    const auto throw_msg_matcher = null_string_matcher;
+
+    test_image_conversion(img_path, expected_img_path, qemuimg_info_output, qemuimg_info_result, attempt_convert,
+                          qemuimg_convert_result, throw_msg_matcher);
+}
+
+TEST(BackendUtils, image_convert_converts_when_raw_image)
+{
+    const auto img_path = "/fake/img/path";
+    const auto expected_img_path = "/fake/img/path.qcow2";
+    const auto qemuimg_info_output = "{\n    \"format\": \"raw\"\n}";
+    const auto qemuimg_info_result = success;
+    const auto attempt_convert = true;
+    const auto qemuimg_convert_result = success;
+    const auto throw_msg_matcher = null_string_matcher;
+
+    test_image_conversion(img_path, expected_img_path, qemuimg_info_output, qemuimg_info_result, attempt_convert,
+                          qemuimg_convert_result, throw_msg_matcher);
+}
+
+TEST(BackendUtils, image_convert_img_not_found_info_fails_throws)
+{
+    const auto img_path = "/fake/img/path";
+    const auto expected_img_path = "/fake/img/path.qcow2";
+    const auto qemuimg_info_output = "not found";
+    const auto qemuimg_info_result = failure;
+    const auto attempt_convert = false;
+    const auto qemuimg_convert_result = mp::ProcessState{};
+    const auto throw_msg_matcher = mp::make_optional(HasSubstr(qemuimg_info_output));
+
+    test_image_conversion(img_path, expected_img_path, qemuimg_info_output, qemuimg_info_result, attempt_convert,
+                          qemuimg_convert_result, throw_msg_matcher);
+}
+
+TEST(BackendUtils, image_convert_converts_fails_throws)
+{
+    const auto img_path = "/fake/img/path";
+    const auto expected_img_path = "/fake/img/path.qcow2";
+    const auto qemuimg_info_output = "{\n    \"format\": \"raw\"\n}";
+    const auto qemuimg_info_result = success;
+    const auto attempt_convert = true;
+    const auto qemuimg_convert_result = failure;
+    const auto throw_msg_matcher = mp::make_optional(HasSubstr("qemu-img failed"));
+
+    test_image_conversion(img_path, expected_img_path, qemuimg_info_output, qemuimg_info_result, attempt_convert,
+                          qemuimg_convert_result, throw_msg_matcher);
+}
```

Comment:
Looks like a parameterized test to me :)

Reply:
Yeah, quite a bit of code duplication here.  I'll see about doing that.

Reply:
Ok, made the tests parameterized.